library/src/amd_detail/hipsolver_sparse.cpp

```diff
@@ -1074,4 +1085,275 @@ catch(...)
     return hipsolver::exception2hip_status();
 }*/
 
+/******************** CSRLSVQR ********************/
+hipsolverStatus_t hipsolverSpScsrlsvqr(hipsolverSpHandle_t       handle,
+                                       int                       n,
+                                       int                       nnz,
+                                       const hipsparseMatDescr_t descrA,
+                                       const float*              csrVal,
+                                       const int*                csrRowPtr,
+                                       const int*                csrColInd,
+                                       const float*              b,
+                                       double                    tolerance,
+                                       int                       reorder,
+                                       float*                    x,
+                                       int*                      singularity)
+try
+{
+    if(!handle)
+        return HIPSOLVER_STATUS_NOT_INITIALIZED;
+    if(n < 0 || nnz < 0)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+    if(!descrA)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+    if(!csrRowPtr || !csrColInd || !csrVal || !descrA)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+    if(!b || !x || !singularity)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+
+    rocsparse_matrix_type mattype = rocsparse_get_mat_type((rocsparse_mat_descr)descrA);
+    rocsparse_index_base  indbase = rocsparse_get_mat_index_base((rocsparse_mat_descr)descrA);
+    if(mattype != rocsparse_matrix_type_general)
+        return HIPSOLVER_STATUS_MATRIX_TYPE_NOT_SUPPORTED;
+    if(indbase != rocsparse_index_base_zero && indbase != rocsparse_index_base_one)
+        return HIPSOLVER_STATUS_MATRIX_TYPE_NOT_SUPPORTED;
+
+    hipsolverSpHandle* sp = (hipsolverSpHandle*)handle;
+    *singularity          = -1;
+
+    // set up B
+    CHECK_HIP_ERROR(hipMemcpy((void*)x, b, sizeof(float) * n, hipMemcpyDeviceToDevice));
+
+    // convert A to dense matrix
+    float* denseA;
+    CHECK_HIP_ERROR(hipMalloc(&denseA, sizeof(float) * n * n));
```

Comment:
Ideally, this would be allocated in the hipsolverSp handle, but converting to dense is a temporary workaround so I'm perfectly fine with keeping it here.